ouroboros-consensus/src/Ouroboros/Consensus/Protocol/PBFT/State.hs

```diff
@@ -577,44 +579,98 @@ fromList k n (anchor, signers, ebbs) =
   Serialization
 -------------------------------------------------------------------------------}
 
-serializationFormatVersion0 :: VersionNumber
-serializationFormatVersion0 = 0
-
-encodePBftState :: Serialise (PBftVerKeyHash c)
+toCompactFormat ::
+     forall c. PBftCrypto c
+  => PBftState c
+  -> ( Map (PBftVerKeyHash c) ([SlotNo], [SlotNo], [SlotNo], [SlotNo])
+       -- For each signer, the slots in 'preAnchor', 'postAnchor',
+       -- 'preWindow', and 'inWindow'.
+     , MaybeEbbInfo
+     )
```

Comment:
Probably overkill, @edsko will likely say so. I reckon he'd just want me to encode the four `StrictSeq (PBftSigner c)`s directly to keep it simple. But ... bytes :grin:

Reply:
I'm so glad you're still having internal dialogues with me even when I'm not around, saves me so much hassle :grimacing: